torch/csrc/autograd/variable.cpp

```diff
@@ -107,19 +105,8 @@ VariableViewImpl::VariableViewImpl(Variable base_, at::Tensor data_, VarFlags fl
 
 std::shared_ptr<Function>& VariableViewImpl::get_grad_fn() {
   std::lock_guard<std::mutex> lock(mutex);
-  if (base.requires_grad() && !requires_grad) {
-    // TODO: See test_inplace_view_flags. It would be good to support this operation
-    // but that might require sharing requires_grad between the base and the view
-    throw std::runtime_error(
-        "requires_grad is False and base.requires_grad is True. Cannot use "
-        "this view in a differentiable operation. Re-create the view from its "
-        "base Variable after the last in-place modification.");
-  }
-  if (base.is_volatile() && !is_volatile) {
-    throw std::runtime_error(
-        "is_volatile is False and base.is_volatile is True. Cannot use "
-        "this view in a differentiable operation. Re-create the view from its "
-        "base Variable after the last in-place modification.");
+  if (!_grad_fn && !base.requires_grad()) {
```

Comment:
I think this breaks when you `detach_()` the base. I think we're end up with a view that doesn't require grad, but the version counter is the same, so views won't even notice and will happily return their old grad_fns. We need to check `self.grad_fn()`

Reply:
Not sure what you expect here: if you modify Tensors through a detached variable it's not going to affect gradients. That was true before and after this PR.

Reply:
I can't see how is this related to what I said. My point is that doing `detach_()` on the base will also cause the views to not require grad, but they will retain their grad functions

Reply:
Detaching the base does not detach views.  Any variable with a non-null `grad_fn` also has `var.requires_grad() = true`:

https://github.com/colesbury/pytorch/blob/d5fa2c1e8af3471fbcd9ccb8d4a125036cd31707/torch/csrc/autograd/variable.h#L260-L262